### PR TITLE
[FE] Display Message to User when Attempting to Pre-Check-In on the Day of Appointment

### PR DESCRIPTION
**Pull request by brdunfield**
## Description
In progress PR to show the user a unique error page / message when attempting to use a pre-check-in link on the same day as their appointment(s).

This PR:
- Checks whether appointments associated to a user are tomorrow or later (with a new helper function). If no valid appointments are found that are tomorrow or later, the user is forwarded to an error page with a new header and message.
- Refactors the Error page to add capability to change the message / header based on a query param in the URL. The props that are sent to ErrorMessage are dependent on what value the `type` param has. currently `expired` shows the new error messages.
- Adds PLACEHOLDER translations for spanish - i'm not sure if we have official translations for this page? If not, should I leave the placeholder translations or put english for both?

How to test:
The existing (and default) UUID '0429dda5-4165-46be-9ed1-1e652a8dfd83', should work exactly the same as before. The appointments using this UUID are now set to tomorrow instead of the current date, so that they remain valid.
A new test UUID has been added: '354d5b3a-b7b7-4e5c-99e4-8d563f15c521'
The appointments for this UUID are all set to the current day. Using this UUID in the link will therefore direct to the new error page.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#37983

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by brdunfield** [comment]
Does this clause
https://github.com/department-of-veterans-affairs/vets-website/pull/20895/files#diff-6023b0b7ff5efdd9caa6d41b4e2c817ae98ad07a0de8c698e72e4fbd280a73d9R36

Still have a path to be accessed, or does this PR replace it (and can I rip it out)?

File: src/applications/check-in/pre-check-in/pages/Error/index.jsx [hunk 1]
```diff
@@ -51,16 +53,34 @@ const Error = () => {
     </>
   );
 
+  // use type param to set unique error messages
+  const getErrorMessagePropsByType = () => {
```
Comment (brdunfield) [inline src/applications/check-in/pre-check-in/pages/Error/index.jsx:57]: If we're going to go with the params method, here is where we could extend Error to pass alternate unique messages to ErrorMessage

File: src/applications/check-in/pre-check-in/pages/Introduction/index.jsx [hunk 2]
```diff
@@ -39,6 +40,9 @@ const Introduction = props => {
 
   useEffect(
     () => {
+      const setDataToState = async data => {
```
Comment (brdunfield) [inline src/applications/check-in/pre-check-in/pages/Introduction/index.jsx:43]: This is not super related to this PR, but it was causing a react error, as it was previously trying to run this code after this page / component unmounts.

Comment (brianseek) [reply src/applications/check-in/pre-check-in/pages/Introduction/index.jsx:43]: nice!

**Review by brianseek — APPROVED** [review]
LGTM!

File: src/applications/check-in/utils/appointment/index.js [hunk 3]
```diff
@@ -72,8 +72,18 @@ const removeTimeZone = payload => {
 
   return updatedPayload;
 };
+
+const preCheckinExpired = appointments => {
```
Comment (acrollet) [inline src/applications/check-in/utils/appointment/index.js:76]: sorry for the drive-by comment, I'm realizing as I look at my own branch  that we need to make sure to add unit tests for any new functions here :-)

Comment (acrollet) [inline src/applications/check-in/utils/appointment/index.js:76]: sorry for the drive-by comment, I'm realizing as I look at my own branch  that we need to make sure to add unit tests for any new functions here :-)
